components/AdBanner/AdBanner.js

```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import OutboundLink from 'components/_common_/OutboundLink/OutboundLink';
-import MockedRouter from 'test-utils/mocks/nextRouterMock';
+import Router from 'next/router';
```

Comment:
In the cases where you imported a router, can you use `withRouter` and decorate the component instead?

Reply:
If you decorate a component with `withRouter` it then has access to router props.

```javascript
import { withRouter } from 'next/router';

class SomeComponent extends React.Component {
  render() {
    const { router } = this.props;
    return <div>{router.pathname}</div>;
  }
}

export default withRouter(SomeComponent)
```

So, you'll decorate AdBanner, and then you'll have access to `router` as a prop in that component which you'll then pass to `OutboundLink`

Reply:
I have used props property and fixed Route error which i was getting in my local build command (yarn build).  Now every command is working in my local except **yarn export**. Not sure why.  So i have raised an issue with next.js [here](https://github.com/zeit/next.js/issues/5637).   

Let's see if they reply something in this regard.